File: sdks/java/lib/src/main/java/models/shared/LegalHoldPolicyAssignmentBase.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * LegalHoldPolicyAssignmentBase
 * Legal Hold Assignments are used to assign Legal Hold
 * Policies to Users, Folders, Files, or File Versions.
 * 
 * Creating a Legal Hold Assignment puts a hold
 * on the File-Versions that belong to the Assignment's
 * 'apply-to' entity.
**/
public class LegalHoldPolicyAssignmentBase {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public LegalHoldPolicyAssignmentBase withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public LegalHoldPolicyAssignmentBaseTypeEnum type;
    public LegalHoldPolicyAssignmentBase withType(LegalHoldPolicyAssignmentBaseTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: python-sdk/src/boxsdk/models/shared/legalholdpolicyassignment_base.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class LegalHoldPolicyAssignmentBaseTypeEnum(str, Enum):
    LEGAL_HOLD_POLICY_ASSIGNMENT = "legal_hold_policy_assignment"


@dataclass_json
@dataclasses.dataclass
class LegalHoldPolicyAssignmentBase:
    r"""LegalHoldPolicyAssignmentBase
    Legal Hold Assignments are used to assign Legal Hold
    Policies to Users, Folders, Files, or File Versions.
    
    Creating a Legal Hold Assignment puts a hold
    on the File-Versions that belong to the Assignment's
    'apply-to' entity.
    """
    
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: Optional[LegalHoldPolicyAssignmentBaseTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
